.github/workflows/wash.yml

```diff
@@ -150,13 +167,22 @@ jobs:
         with:
           language: ${{ matrix.template.lang }}
           language-version: ${{ matrix.template.lang-version }}
-          wash-version: ${{ matrix.wash-version }}
+          wash-version: ${{ matrix.wash-version != 'current' && matrix.wash-version || '' }}
 
       # Set up Rust & trigger cargo toolchain
       - run: rustup show
-      - uses: Swatinem/rust-cache@82a92a6e8fbeee089604da2575dc567ae9ddeaab
+      - name: Download wash artifact
+        if: ${{ matrix.wash-version == 'current' }}
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16
         with:
-          shared-key: 'ubuntu-22.04-shared-cache'
+          name: wash
+      # Cargo bin should exist and be in the PATH already so we just move it there to make our lives simple
+      - name: Move wash artifact to cargo bin
+        if: ${{ matrix.wash-version == 'current' }}
+        run: |
+          mkdir -p ~/.cargo/bin
+          chmod +x ./wash
+          mv ./wash ~/.cargo/bin/wash
```

Comment:
can remove this whole section since wash gets installed in `wasm-lang-toolchain-setup`